File: CS50x-2024-Pset01-C/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int minCoins(int change);

int main(void)
{
    int change;
    do
    {
        change = get_int("Change owed: ");
    }
    while (change < 0);

    printf("%d\n", minCoins(change));
    return 0;
}

int minCoins(int change)
{
    int num25, num10, num5, num1;
    if (change == 0)
    {
        return 0;
    }
    num25 = change / 25;
    change -= num25 * 25;
    num10 = change / 10;
    change -= num10 * 10;
    num5 = change / 5;
    change -= num5 * 5;
    num1 = change;
    change = 0;
    return num25 + num10 + num5 + num1;
}

File: CS50x-2024-Pset06-Python/sentimental-cash/cash.py
from cs50 import get_float

while True:
    try:
        change = get_float("Changes: ")
        if change >= 0:
            break
    except ValueError:
        pass


change = int(change * 100)
min_coins = 0

for coin_value in [25, 10, 5, 1]:
    if change <= 0:
        break
    else:
        min_coins += change // coin_value
        change -= coin_value * (change // coin_value)


print(min_coins)
